awx/ui_next/src/components/PaginatedTable/PaginatedTable.jsx

```diff
@@ -0,0 +1,194 @@
+import PropTypes from 'prop-types';
+import { TableComposable, Tbody } from '@patternfly/react-table';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { useHistory } from 'react-router-dom';
+
+import ListHeader from '../ListHeader';
+import ContentEmpty from '../ContentEmpty';
+import ContentError from '../ContentError';
+import ContentLoading from '../ContentLoading';
+import Pagination from '../Pagination';
+import DataListToolbar from '../DataListToolbar';
+
+import {
+  encodeNonDefaultQueryString,
+  parseQueryString,
+  replaceParams,
+} from '../../util/qs';
+import PaginatedTableRow from './PaginatedTableRow';
+import { QSConfig, SearchColumns } from '../../types';
+
+function PaginatedTable({
+  contentError,
+  hasContentLoading,
+  emptyStateControls,
+  items,
+  itemCount,
+  qsConfig,
+  headerRow,
+  renderRow,
+  toolbarSearchColumns,
+  toolbarSearchableKeys,
+  toolbarRelatedSearchableKeys,
+  pluralizedItemName,
+  showPageSizeOptions,
+  i18n,
+  renderToolbar,
+  // onRowClick,
+}) {
+  const history = useHistory();
+
+  // const handleListItemSelect = (id = 0) => {
+  //   const match = items.find(item => item.id === Number(id));
+  //   onRowClick(match);
+  // };
+
+  const pushHistoryState = params => {
+    const { pathname } = history.location;
+    const encodedParams = encodeNonDefaultQueryString(qsConfig, params);
+    history.push(encodedParams ? `${pathname}?${encodedParams}` : pathname);
+  };
+
+  const handleSetPage = (event, pageNumber) => {
+    const oldParams = parseQueryString(qsConfig, history.location.search);
+    pushHistoryState(replaceParams(oldParams, { page: pageNumber }));
+  };
+
+  const handleSetPageSize = (event, pageSize, page) => {
+    const oldParams = parseQueryString(qsConfig, history.location.search);
+    pushHistoryState(replaceParams(oldParams, { page_size: pageSize, page }));
+  };
+
+  const searchColumns = toolbarSearchColumns.length
+    ? toolbarSearchColumns
+    : [
+        {
+          name: i18n._(t`Name`),
+          key: 'name',
+          isDefault: true,
+        },
+      ];
+  const queryParams = parseQueryString(qsConfig, history.location.search);
+
+  const dataListLabel = i18n._(t`${pluralizedItemName} List`);
+  const emptyContentMessage = i18n._(
+    t`Please add ${pluralizedItemName} to populate this list `
+  );
+  const emptyContentTitle = i18n._(t`No ${pluralizedItemName} Found `);
+
+  let Content;
+  if (hasContentLoading && items.length <= 0) {
+    Content = <ContentLoading />;
+  } else if (contentError) {
+    Content = <ContentError error={contentError} />;
+  } else if (items.length <= 0) {
+    Content = (
+      <ContentEmpty title={emptyContentTitle} message={emptyContentMessage} />
+    );
+  } else {
+    Content = (
+      <TableComposable aria-label={dataListLabel}>
+        {headerRow}
+        <Tbody>{items.map(renderRow)}</Tbody>
+      </TableComposable>
+    );
+  }
+
+  const ToolbarPagination = (
```

Comment:
This seems to be a duplicate of the same component that starts[here](https://github.com/ansible/awx/pull/8729/files#diff-b942f30c50317e77ddaf815b6d7e3bbe508ea5fa74e43318eee5a31050af5f57R135) with a few exceptions.  Could they be combined?